.github/workflows/publish_docker_images_on_tag.yml

```diff
@@ -0,0 +1,47 @@
+name: Build Docker Image On Tag Push
+
+env:
+  DOCKERHUB_USER_NAME: ${{ secrets.DOCKERHUB_USER_NAME }}
+  DOCKERHUB_PASSWORD: ${{ secrets.DOCKERHUB_PASSWORD }}
+
+on:
+  push:
+    tags:
+      - "**"
+jobs:
+  build_and_publish_images:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        image_type:
+          - latest
+          - alpine
+          - postgres_12
+          - postgres_13
+          - nightly
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          ref: master
+
+      - name: Clone tools branch
+        run: git clone -b v0.8.11 --depth=1 https://github.com/citusdata/tools.git tools
```

Comment:
I wonder --one day-- if we should start using the master branch itself to stop dealing with versioning in tools repo; we anyway do not merge develop into master until making sure that all the changes are ok.

Reply:
I do not think that it would be possible soon. We usually do breaking changes, and we do not want to break some infrequent stuff such as `hll`/`topn` packaging for example.

Reply:
I agree with @hanefi . We need to continue using tags to make our tools repo usage safe. Since we are using tools like a package, we need a versioning mechanism which is tag here